Sources/tasks/CreatePost-Notify.php

```diff
@@ -280,7 +280,7 @@ public function execute()
 			$smcFunc['db_query']('', '
 				UPDATE {db_prefix}log_notify
 				SET sent = {int:is_sent}
-				WHERE (id_topic = {int:topic} OR id_board = {int:board})
+				WHERE ' . ($type == 'topic' ? 'id_board = {int:board}' : 'id_topic = {int:topic}') . '
```

Comment:
Just checking, but did you actually mean this instead?
```suggestion
				WHERE ' . ($type !== 'topic' ? 'id_board = {int:board}' : 'id_topic = {int:topic}') . '
```
It seems odd to mark the board notification as sent when the notification type was topic.

Reply:
Odd perhaps, but is a new topic notif; therefore id_topic won't be set.

https://github.com/SimpleMachines/SMF/blob/6060f363db575314b4c59a9a20485a1081439fbb/Sources/Subs-Post.php#L2170

Reply:
@Sesquipedalian - this is correct.  The internal naming of the types can be a bit confusing...  But a board watch triggers a 'topic' notification type (i.e., upon a new topic) and a topic watch triggers a 'reply' notification type.  So when you see a 'topic' notification, you gotta look up the board to set the sent flag.

This very confusion was at the heart of some of the 2.1.3 notification PRs...  This one wasn't caught then.

Reply:
Cross post...

Reply:
Fair enough. Thanks for the explanation. 🙂